guides/image-sources/provisioning-image-cloudinit.plantuml

```diff
@@ -0,0 +1,55 @@
+@startuml
+
+!include foreman.pstyle
+
+title Image-based provisioning with cloud-init configuration
+
+actor       User
+participant "Provisioned\nInstance" as Host
+participant "Foreman" as Foreman
+participant "Foreman\nProxy" as Proxy
+participant "Infrastructure\nCloud" as Cloud
+participant DNS
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Foreman : has an image\nwith cloud-init flag
+
+== Create host in Foreman ==
+
+User -> Foreman : clicks **Create Host**
+User -> Foreman : selects cloud resource,\nthe image and submits
+group Template [cloud-init]
+   Foreman -> Proxy : renders the cloud-init script
+end
+Foreman -> Cloud : creates new instance\nwith cloud-init info
+Cloud -> Foreman : reports IP address
+Foreman -> Proxy : creates DNS records
+Proxy -> DNS : forwards DNS records
+!if ($puppet)
+Foreman -> Proxy : creates Puppet sign request (autosign enabled)
+Proxy -> Puppet : forwards Puppet sign request
+!endif
+Foreman -> Cloud : starts the instance
```

Comment:
Referencing the code: https://github.com/theforeman/foreman/blob/e80d5558767bb713c13264e821122baea577af76/app/models/concerns/orchestration/compute.rb#L79-L82

Reply:
I'm assuming that means that it's correct.